src/hostapi/pulseaudio/pa_linux_pulseaudio_cb.c

```diff
@@ -119,12 +119,15 @@ void PaPulseAudio_ReleaseOperation(PaPulseAudio_HostApiRepresentation *hostapi,
 
     while( waitOperation > 0 )
     {
-        PaPulseAudio_Lock( hostapi->mainloop );
-        pa_threaded_mainloop_wait( hostapi->mainloop );
-        PaPulseAudio_UnLock( hostapi->mainloop );
-
         localOperationState = pa_operation_get_state( localOperation );
```

Comment:
This is still racy, because the pulseaudio server could release the stream in the time span between pa_operation_get_state() and pa_threaded_mainloop_wait() and you'd be back to square one with a hang.

Instead I'd do...

```
PaPulseAudio_Lock( hostapi->mainloop );
localOperationState = pa_operation_get_state( localOperation );
if( localOperationState == PA_OPERATION_RUNNING )
{
    pa_threaded_mainloop_wait( hostapi->mainloop );
}
 PaPulseAudio_UnLock( hostapi->mainloop );
```

... to avoid this race.

Reply:
Thank you taking look at this. This will fix this situation for good.

Reply:
As a little optimization, I'd also call `localOperationState = pa_operation_get_state( localOperation );` again, immediately after the `pa_threaded_mainloop_wait( hostapi->mainloop );` so that the following
`if( localOperationState != PA_OPERATION_RUNNING)` will decide if it should break; out of the while loop, based on the latest localOperationsState.

This way you could skip a potentially not needed usleep(1000); and avoid a needless 1 millisecond delay when this function is called, which seems to be from various places.

Reply:
Ok I've made bit optimization. Removed delay as it sleep in semaphore in `pa_threaded_mainloop_wait` if there comes signals very quick they should come that quick they smack down CPU.